runfolder_tests/run_integration_tests.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/python
+import socket
+import subprocess
+import os
+import time
+import sys
+import shutil
+import stat
+
+class IntegrationTestHelper:
+    @staticmethod
+    def find_port():
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(('localhost', 0))
+        _, port = s.getsockname()
+        s.close()
+        return port
+
+    @staticmethod
+    def is_listening(port):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        result = sock.connect_ex(('localhost', port))
+        return result == 0
+
+    def wait_for_listening(self, port):
+        while not self.is_listening(port):
+            print "waiting"
+            time.sleep(0.1)
+
+
+def execute(args, script_name=None):
+    """
+    Execute the args using Popen, but create and execute a corresponding
+    script if script_name is provided
+    """
+    if script_name:
+        with open(script_name, "w") as f:
+            f.write("#!/bin/bash\n")
+            f.write(" ".join(args))
+            f.write("\n")
+        os.chmod(script_name, stat.S_IRWXU)
+        service = subprocess.Popen([os.path.join(".", script_name)])
+    else:
+        service = subprocess.Popen(args)
+    return service
+
+
+def setup_testrun_dir():
+    """
+    Sets up a testrun_* directory in the cwd and returns the path to it
+    """
+    test_run = "testrun_{}".format(int(time.time()))
+    os.mkdir(test_run)
+    this_files_dir = os.path.dirname(os.path.realpath(__file__))
+    config_templates = os.path.join(this_files_dir, "integration", "config")
+    os.mkdir(os.path.join(test_run, "runfolders"))
+    shutil.copy2(os.path.join(config_templates, "app.config"), test_run)
+    shutil.copy2(os.path.join(config_templates, "logger.config"), test_run)
+    return os.path.realpath(test_run)
+
+
+def setup_local_server(port, dir):
```

Comment:
`dir()` is a reserved word for python: https://docs.python.org/2/library/functions.html so maybe rename the param to directory?